### DISCUSSION TITLE
Need to Retry only those tests which failed with CypressError and not those test which are failed with AssertionError

### DISCUSSION THREAD

**Discussion by ghost** [cypress-io/cypress#17421] [General]
I am using Cypress Cucumber framework and i want to retry only those test cases which are failed due to CypressError in aftereach hooks but i am not able to do that for the workaround what i did is 

I have cypress.json  in which i mention the retries: 1 before tests starts
and now lets suppose my test got failed due to AssertionError and it landed in aftereach hooks here i am updating retries: 0 in the cypress.json file on runtime by using cy.writeFile('cypress.json', 'wholedatastring') the value in retries got updated to 0 but even then cypress is attempting the retry that failed test

if (this.currentTest.err.name == 'AssertionError') {
          cy.log('No Need to retry')
          cy.readFile('cypress.json').then(text => {
            console.log('json coming')
            console.log(JSON.stringify(text.retries))
            text.retries=0
            cy.writeFile('cypress.json', text)
          })
So my question is if i am updating the value of retries in cypress.json and it also got updated then why cypress is considering the retries as 1 which is before update in aftereach hooks

**Comment by srp-at-ah** [comment]
In answering the question "can I access the attempts count?" (see https://docs.cypress.io/guides/guides/test-retries#Can-I-access-the-current-attempt-counter-from-the-test) the docs say "although ordinarily you would not have to, since this is a low-level detail.". Unfortunately nowhere I see that it is advised or possible to perform any arithmetic on it. I assume 'impossible' and my personal advice would be to find a different approach to solving the problem you are facing, an approach that does not involve modifying low-level details at runtime.
